### ISSUE TITLE
TypeError: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by adamlacombe** [ionic-team/stencil#860]
**Stencil version:**

```
 @stencil/core@0.9.7
```

**I'm submitting a:**

[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**

When using the @Method() decorator on a async function the build fails.

**Expected behavior:**

Build wouldn't fail.

**Steps to reproduce:**

**Related code:**

```tsx
 @Method()
  async checkLoginStatus() {
    return this.loggedIn = await database.meta().get("user_id") != '';
  }
```

**Other information:**

> TypeError: Cannot read property 'kind' of undefined at
Object.isImportClause
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/typescript/lib/typescript.js:13650:21)
at sourceFileObj.statements.find.st
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6097:16)
at Array.find (native) at getTypeReferenceLocation
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6095:60)
at getAllTypeReferences.reduce
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6059:29)
at Array.reduce (<anonymous>) at getAttributeTypeInfo
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6058:10)
at classNode.members.filter.reduce
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6721:33)
at Array.reduce (<anonymous>) at getMethodDecoratorMeta
(/home/adam/Documents/projects/apps/stencil-ionic-app/node_modules/@stencil/core/dist/compiler/index.js:6708:10)

**Comment by jthoms1** [comment]
Could you provide more code from the Component? I am unable to reproduce with the code provided.

**Comment by jthoms1** [comment]
Just spoke with @adamdbradley on this and he has experienced from a side effect based import.  He had put in a fix so should be good in release `0.9.8`. Thanks!
